python-checks/src/main/java/org/sonar/python/checks/BooleanCheckNotInvertedCheck.java

```diff
@@ -120,8 +122,36 @@ private static void createQuickFix(IssueWithQuickFix issue, String oppositeOpera
   }
 
   private static PythonTextEdit getReplaceEdit(BinaryExpression toUse, Expression toReplace, String oppositeOperator) {
-    List<Tree> children = toReplace.parent().parent().children();
+    // To be able to remove the 'not' sign and the possible space afterwards
+    Tree notAncestor = Optional.ofNullable(TreeUtils.firstAncestorOfKind(toReplace, Kind.NOT))
+      .orElseThrow(() -> new IllegalArgumentException("A NOT tree must be found. It was not the case."));
+
+    List<Tree> children = notAncestor.children();
     return PythonTextEdit.replaceRange(children.get(0), children.get(children.size() - 1),
-      toUse.leftOperand().firstToken().value() + " " + oppositeOperator + " " + toUse.rightOperand().firstToken().value());
+      getNewExpression(toUse, oppositeOperator));
+  }
+
+  private static String getNewExpression(BinaryExpression toUse, String oppositeOperator) {
+    return getText(toUse.leftOperand()) + " " + oppositeOperator + " " + getText(toUse.rightOperand());
+  }
+
+  private static String getText(Tree tree) {
+    if (tree.children().isEmpty()) {
+      if (tree.is(Kind.EXPRESSION_LIST)) {
+        return "";
+      }
+      return ((Token) tree).value();
+    }
+    // Regular argument and tuple may have children
+    if (tree.children().size() == 1 && !tree.children().get(0).is(Kind.REGULAR_ARGUMENT, Kind.TUPLE, Kind.PARENTHESIZED)) {
+      return tree.firstToken().value();
+    }
```

Comment:
If I remove those lines, tests are still green: is this condition really needed?

Reply:
Indeed, it was a remnant of another logic I used: I was returning the value of the tree directly if there was only one child. But if we only return the value when there is no child, as you suggested, those lines are useless